src/app/core/services/ess-upload-file.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { AppConfigService } from './app-config.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EssUploadFileService {
+  private validEncs: string[];
```

Comment:
I question why `EssUploadFileService` and `EssUploadFileComponent` both have the same list of valid ENCs . It seems confusing to me.

Could you please consider removing `validEncs` member from `EssUploadFileService` and then combine `extractEncsFromFile`, `setValidENCs`, and `getValidEncs` into one method called `getValidEncs`

Reply:
valid enclist in EssUploadFileService is defined so that it can be used in further / any components whenever required.
valid enclist in EssUploadFileComponent is defined to use it Component level.